lib/std/os/windows.zig

```diff
@@ -1493,6 +1493,20 @@ pub fn VirtualFree(lpAddress: ?LPVOID, dwSize: usize, dwFreeType: DWORD) void {
     assert(kernel32.VirtualFree(lpAddress, dwSize, dwFreeType) != 0);
 }
 
+pub const VirtualProtectError = error{
+    InvalidAddress,
+    Unexpected,
+};
+
+pub fn VirtualProtect(lpAddress: ?LPVOID, dwSize: SIZE_T, flNewProtect: DWORD, lpflOldProtect: *DWORD) VirtualProtectError!void {
+    if (kernel32.VirtualProtect(lpAddress, dwSize, flNewProtect, lpflOldProtect) == 0) {
+        switch (kernel32.GetLastError()) {
+            .INVALID_ADDRESS => return error.InvalidAddress,
```

Comment:
Is this the only error code that gets returned? If so, that's very surprising!

Reply:
Well, here's the fun of it: MS provide zero documentation on what error codes this can return, and I couldn't really find anything online either! From some random old forum posts, this is the main one I saw referenced, and skimming through the ReactOS code (admittedly sleep-deprived and on mobile) I didn't see any path that returned anything else, but I might be missing something - I can take another look if you'd like, or if you have a better way to figure out the possible errors that'd be fantastic :P

Reply:
I usually look up the impl in ReactOS too. Perhaps Wine will have more to offer on the other hand?

Reply:
Possibly - I'll take another dig through both when I have time

Reply:
ReactOS has more error but most of them don't seem to have windows equivalents.